miniapps/mtop/mtop_integrators.hpp

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MTOPINTEGRATORS_HPP
+#define MTOPINTEGRATORS_HPP
+
+#include "mfem.hpp"
+#include <map>
+
+namespace mfem {
+
+
+class BaseQFunction
```

Comment:
As this will be the only example of a working `ParametricBNLFormIntegrator`, I think this auxilliary `BaseQFunction` class unnecessarily complicates things.

Reply:
It will be used for the topology optimization miniapp. Otherwise one would have to implement a new version and may be new interface for every new physics.